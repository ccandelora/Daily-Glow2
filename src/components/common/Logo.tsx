import React from 'react';
import { Image, StyleSheet, View, ImageStyle, ViewStyle, TextStyle } from 'react-native';
import { Typography } from './Typography';
import theme from '@/constants/theme';

interface LogoProps {
  size?: 'small' | 'medium' | 'large';
  showText?: boolean;
  style?: ViewStyle;
  imageStyle?: ImageStyle;
  textStyle?: TextStyle;
}

export const Logo: React.FC<LogoProps> = ({
  size = 'medium',
  showText = true,
  style,
  imageStyle,
  textStyle,
}) => {
  // Define sizes for different variants
  const sizes = {
    small: { image: 24, text: 16 },
    medium: { image: 60, text: 24 },
    large: { image: 90, text: 36 },
  };

  const selectedSize = sizes[size];

  return (
    <View style={[styles.container, style]}>
      <Image
        source={require('../../../assets/default_transparent_353x345.png')}
        style={[
          styles.logo,
          { width: selectedSize.image, height: selectedSize.image },
          imageStyle,
        ]}
        resizeMode="contain"
      />
      {showText && (
        <Typography
          variant={size === 'large' ? 'h1' : size === 'medium' ? 'h2' : 'h3'}
          style={{
            ...styles.text,
            fontSize: selectedSize.text,
            ...textStyle,
          }}
          glow="medium"
          color={theme.COLORS.primary.green}
        >
          Daily Glow
        </Typography>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  logo: {
    marginRight: theme.SPACING.xs,
  },
  text: {
    textAlign: 'center',
  },
}); 