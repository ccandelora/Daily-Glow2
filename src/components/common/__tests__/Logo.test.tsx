import React from 'react';
import { render } from '@testing-library/react-native';
import { Logo } from '../Logo';

// Mock the Image component to avoid dealing with assets in tests
jest.mock('react-native/Libraries/Image/Image', () => {
  const MockImage = (props: any) => {
    return null;
  };
  MockImage.getSize = jest.fn();
  MockImage.prefetch = jest.fn();
  MockImage.queryCache = jest.fn();
  return MockImage;
});

describe('Logo', () => {
  it('renders with default props', () => {
    const { toJSON } = render(<Logo />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with small size', () => {
    const { toJSON } = render(<Logo size="small" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with large size', () => {
    const { toJSON } = render(<Logo size="large" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('applies custom container style', () => {
    const { toJSON } = render(
      <Logo style={{ marginTop: 10, backgroundColor: 'red' }} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('applies custom image style', () => {
    const { toJSON } = render(
      <Logo imageStyle={{ marginRight: 20, tintColor: 'blue' }} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
}); 