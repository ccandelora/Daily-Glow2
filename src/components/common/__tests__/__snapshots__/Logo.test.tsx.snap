// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo applies custom container style 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      },
      {
        "backgroundColor": "red",
        "marginTop": 10,
      },
    ]
  }
/>
`;

exports[`Logo applies custom image style 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;

exports[`Logo renders with default props 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;

exports[`Logo renders with large size 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;

exports[`Logo renders with small size 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;
